week9 code review

amm0124 opened the pull request:
# week9 코드리뷰 merge입니다!

궁금증 
1) 요청을 하면 응답이 2번 온다는 경우가 존재한다고 하는데, security 부근에서 일어나는 문제 같다고 하는데 왜 그런지 질문 드립니다!
2) test code 작성 시 팁?!을 듣고 싶습니다. 커버리지 툴을 도입하자는 의견이 나왔는데 이 부분에 대해서도 어떤지 궁금합니다!
3) 품목을 구매할 때 (사전주문->외부 api 결제 -> 주문 완료) 형태로 구현하려고 하는데, 주문 완료 메서드를 transaction으로 처리하려고 했습니다. 이 때 동시성 처리는 어떻게 처리해야 하는지 질문을 드리고 싶습니다!

Commit:
DOCS : 버전 명시

amm0124 committed:
1주차 최종 코드 (#7)

* ExceptionAdvice 정의 (#4)

* build: lombok 추가
#3

* chore: build.gradle 정리
#3

* chore: ApplicationError 정의
#3

* chore: BusinessError 정의
#3

* feat: ExceptionAdvice 정의
- BusinessError
- ApplicationError
#3

* chore: exception -> error 패키지명 수정
#3

* chore: error logging 방식 수정
- [] : {}
#3

* DOCS : PR template 업데이트

---------

Co-authored-by: 정지민 <108014449+stopmin@users.noreply.github.com>
Co-authored-by: amm0124 <amm012450@gmail.com>

amm0124 committed:
Merge pull request #8 from kakao-tech-campus-2nd-step3/develop

* ExceptionAdvice 정의 (#4)

* build: lombok 추가
#3

* chore: build.gradle 정리
#3

* chore: ApplicationError 정의
#3

* chore: BusinessError 정의
#3

* feat: ExceptionAdvice 정의
- BusinessError
- ApplicationError
#3

* chore: exception -> error 패키지명 수정
#3

* chore: error logging 방식 수정
- [] : {}
#3

* DOCS : PR template 업데이트

---------

Co-authored-by: 정지민 <108014449+stopmin@users.noreply.github.com>
Co-authored-by: amm0124 <amm012450@gmail.com>

jjt4515 committed:
feat: Member 엔티티 생성 #11

jjt4515 committed:
feat: MemberProfile 엔티티 생성 #11

jjt4515 committed:
feat: LoginType 및 MemberType 생성 #11

jjt4515 committed:
feat: MemberProfile 멤버 변수 추가 #11

stopmin committed:
feat: Farm Entity 생성
#2

stopmin committed:
feat: Farm Entity 수정
- 빌더패턴 적용
- 상태를 나타내는 값 설정
#2

stopmin committed:
feat: FarmManageService 정의
- 등록 인터페이스만 정의
#2

stopmin committed:
feat: FarmRegisterRequest 생성
#2

stopmin committed:
feat: FarmStatus 정의
- 회의 따로 필요함.
#2

stopmin committed:
feat: FarmMangeController 생성
- 등록 API 생성
#2

stopmin committed:
feat: FarmRepository 생성
#2

stopmin committed:
feat: FarmMangeController 수정
- ResponseEntity 리턴한다
#2

stopmin committed:
feat: FarmMangeController 수정
- ResponseEntity 리턴한다
#2

stopmin committed:
Merge remote-tracking branch 'origin/feature/ISSUE-2' into feature/ISSUE-2

# Conflicts:
#	src/main/java/poomasi/domain/farm/controller/FarmMangeController.java

stopmin committed:
feat: FarmResponse 생성
#2

stopmin committed:
feat: 농장 단건 조회 API 생성
#2

stopmin committed:
refactor: 함수 네이밍 수정
- getFarm -> getFarmByFarmId
#2

jjt4515 committed:
feat: 일반 로그인 구현 #11

jjt4515 committed:
feat: 로그인 에러 처리 #11

jjt4515 committed:
docs: yml 파일 작성 #11

stopmin committed:
refactor: 판매자 & 일반 유저 API 분리
- Seller: 판매자
#2

stopmin committed:
refactor: 판매자 & 일반 유저 API 분리
- Seller: 판매자
#2

stopmin committed:
Merge remote-tracking branch 'origin/feature/ISSUE-2' into feature/ISSUE-2

# Conflicts:
#	src/main/java/poomasi/domain/farm/controller/FarmController.java
#	src/main/java/poomasi/domain/farm/controller/FarmSellerController.java

jjt4515 committed:
feat: Application 코드 어노테이션 추가 #11

jjt4515 committed:
refactor: JwtUtil을 JwtProvider로 변경 및 코드 추가 #11

jjt4515 committed:
refactor: MemberType 재정의 #11

canyos committed:
feat: product 생성 삭제 수정

canyos committed:
feat: category crud

jjt4515 committed:
refactor: MemberType -> Role 이름 변경 #11

stopmin committed:
refactor: seller -> farmer
#2

stopmin committed:
refactor: endpoint 수정
#2

stopmin committed:
chore: 주석 제거
#2

stopmin committed:
feat: 농장 설명 추가
#2

stopmin committed:
feat: application-secret.yml ignore
#2

stopmin committed:
build: mysql 종속성 추가
- 8.0.33
#2

stopmin committed:
feat: 농장 목록 조회
#2

stopmin committed:
feat: 예시 에러 삭제
#2

stopmin committed:
feat: 농장 업데이트
#2

stopmin committed:
feat: FarmResponse 수정
- description 추가
#2

jjt4515 committed:
docs: application-secret 작성 #11

jjt4515 committed:
feat: Redis 설정 및 config 작성 #11

jjt4515 committed:
feat: RedisService 코드 작성 #11

jjt4515 committed:
feat: RefreshToken 작성 #11

jjt4515 committed:
feat: RefreshTokenService 작성 #11

stopmin committed:
Merge pull request #19 from kakao-tech-campus-2nd-step3/feature/ISSUE-2

Farm 엔티티를 개발한다

jjt4515 committed:
feat: removeMemberRefreshToken 및 getTokenResponse 구현 #11

jjt4515 committed:
refactor: Member 멤버변수 snsAuthId -> kakaoAuthId 로 수정 #11

jjt4515 committed:
feat: 구매자 로컬 회원가입, 로그아웃 구현 #11

jjt4515 committed:
refactor: 회원가입 로직 수정 #11

jjt4515 committed:
feat: 농부 승격 로직 구현 #11

canyos committed:
refactor: 지민님 피드백 수정

jjt4515 committed:
refactor: utf-8명시 #11

jjt4515 committed:
refactor: 회원가입 endpoint명 변경 #11

jjt4515 committed:
refactor: 토큰 유효성 검사 에러 로그 단순화 #11

jjt4515 committed:
로그인 삭제 및 BusinessError 구분 #11

jjt4515 committed:
농부 자격 증명 에러 수정 #11

jjt4515 committed:
refactor: 멤버 소프트 딜리트 쿼리 수정 #11

jjt4515 committed:
refactor: MemberProfile의 isSuspended를 isBanned로 네이밍 변경 #11

jjt4515 committed:
refactor: MemberService 리팩토링 #11

jjt4515 committed:
chore: 회원가입, 로그아웃, 농부 승격 auth 폴더로 이동 #11

jjt4515 committed:
chore: LoginRequest 위치 변경 #11

jjt4515 committed:
Merge branch 'week4' into feature/ISSUE-11

jjt4515 committed:
Merge pull request #20 from kakao-tech-campus-2nd-step3/feature/ISSUE-11

멤버 인증 인가

canyos committed:
refactor: 상품 enum 삭제

canyos committed:
Merge branch 'week4' into feature/ISSUE-17

stopmin committed:
chore: 린트 적용 및 잘못된 수정사항 제거
#17

jjt4515 committed:
refactor: 회원가입 엔드포인트 수정, .gitignore 파일 수정 #21

jjt4515 committed:
feat: Redis 블랙리스트 로직 추가 #21

jjt4515 committed:
feat: AuthService 로그아웃 로직 수정 #21

jjt4515 committed:
feat: AuthController 로그아웃 로직 수정 #21:

jjt4515 committed:
feat: JwtProvider 토큰 유효성 검사 로직 수정 #21

jjt4515 committed:
refactor: AuthService 리팩토링 #21

jjt4515 committed:
refactor: RefreshToken 유효성 검사 로직 수정 #21

jjt4515 committed:
refactor: RefreshToken 커스텀 에러 수정 #21

jjt4515 committed:
refactor: RedisService 리팩토링 #21

jjt4515 committed:
refactor: JwtUtil 리팩토링 #21

jjt4515 committed:
refactor: 농부로 승격 로직 위치 변경 #21

jjt4515 committed:
refactor: 사용자 리프레시 토큰 제거 로직 리팩토링 #21

jjt4515 committed:
AuthService 및 RefreshTokenService 리팩토링 #24

jjt4515 committed:
refactor: Jwt claims에 role과 email이 들어가도록 수정 #24

jjt4515 committed:
refactor: 시크릿 파일 삭제 #21

jjt4515 committed:
refactor: 시크릿 파일 삭제 #21

jjt4515 committed:
Merge pull request #23 from kakao-tech-campus-2nd-step3/feature/ISSUE-21

Redis 수정

jjt4515 committed:
Merge pull request #25 from kakao-tech-campus-2nd-step3/feature/ISSUE-24

JwtUtil 수정

stopmin committed:
refactor: 어노테이션 일부 수정
- RequiredArgsConstructor
- Comment
- SQLSelect 삭제
#13

canyos committed:
refactor: 2차 수정
RequiredArgsConstructor 사용
PathVariable camel case로 변경

canyos committed:
Product, Category Merge

Product, Category Entity

Commit:
feat : Member Entity 생성자 추가

Commit:
refactoring : ROLE 객체 get Authority(String) 메서드 추가 (Overriding)

Commit:
feat : security filter chain을 위한 config

Commit:
feat : User detail 추가

Commit:
feat : security에서 사용할 객체 spring bean으로 등록

Commit:
feat : username password 기반 로그인 필터

Commit:
feat : jwt 인증 필터

Commit:
build : oauth2.0 client 의존성 추가

stopmin committed:
농장 예약 가능 시간대 등록 및 조회 (#30)

* feat: FarmSchedule 추가
#13

* refacto: userId로 네이밍 수정
#13

* feat: secret 내리자...

* feat:FarmSchedule 도메인 추가
- 스케쥴 추가
- 스케쥴 조회
#13

* refactor: AllArgsConstructor -> RequiredArgsConstructor
#13

* refactor: Enumerated추가

* refactor: where/sqlselect 제거

* feat: 이미 농장이 있는 경우 새로운 농장 등록을 막는다

* bug: 농장 조회 기능 에러 수정
- 쿼리로 삭제된 농장 조회 필터링
-  JPA 메소드 수정
#13

* bug: 농장 조회 기능 에러 수정
- 쿼리로 삭제된 농장 조회 필터링
-  JPA 메소드 수정
#13

* feat: 농장 스케쥴 등록
- 시작 날짜, 종료 날짜, 요일 선택시 추가 가능
#13

* feat: 농장 예약 가능날짜 조회
#13

* chore: 필요없는 코드 정리
#13

* chore: 필요없는 코드 정리
#13

* refactor: schedule 도메인 분리
- 생각보다 너무 커져서 Farm 안에 집어 넣어버렸습니다
#13

* refactor: schedule controller 분리
#13

* refactor: schedule controller 분리
#13

* refactor: now() -> current_timestamp
#13

* refactor: userId -> memberId
#13

* refactor: 기간 범위의 스케쥴 한번에 가져와서 비교하도록 수정
#13

* feature: 체험 비용 추가
#13

* feat: 농장 스케쥴 등록 validation 추가
#36

* feat: 농장 스케쥴 등록 validation 추가
#36

* 충돌 해결

canyos committed:
feat: product review 생성 조회

stopmin committed:
@Where, @SQLSelect 어노테이션을 JPA 방식으로 변환 (#38)

* refactor: category 위치 이동
#36

* refactor: RequestMapping으로 묶음
#36

* refactor: Member 도메인 일부 수정
- SQLSelect 삭제
- isDeleted -> deletedAt
#36

* feature: 상품 등록 시 validation
- 농부인가?
- 존재하는 카테고리인가?
#36

* feature: 상품 단건 조회 & 다건 조회
#36

* feature: 상품 단건 조회 & 다건 조회
#36

* refactor: 요청 방식 수정
- 리소스 기반으로 표현
#36

* refactor: 요청 방식 수정
- 리소스 기반으로 표현
#36

* refactor: 파일 위치 수정
#36

* refactor: 파일 위치 수정
#36

* refactor: 중복코드 삭제
#36

* fix: 수량 업데이트 동작 안하는 부분 수정
#36

* feature: 상품 단건 조회
- 나중에 DTO는 바껴야할듯
#36

* feature: 회원 조회 조건 수정
- DeletedAtIsNull
#36

amm0124 committed:
Merge branch 'week4' into feature/ISSUE-22

amm0124 committed:
Merge pull request #34 from kakao-tech-campus-2nd-step3/feature/ISSUE-22

username / jwt 인증 filter 구현

canyos committed:
feat: category - product 연관관계 추가

canyos committed:
feat: 카테고리 내 상품 조회

canyos committed:
fix: 빠진 파일 추가

canyos committed:
style: 아 맞다 정렬

canyos committed:
feat: 상품리뷰 - 상품리뷰사진 연관 관계 생성

canyos committed:
refactor: 평균 평점을 리뷰 추가할 때 계산하도록 변경

amm0124 committed:
Merge pull request #44 from kakao-tech-campus-2nd-step3/week4

Week4 머지

amm0124 committed:
Merge pull request #45 from kakao-tech-campus-2nd-step3/develop

master merge

stopmin committed:
Merge branch 'week5' into feature/ISSUE-29

# Conflicts:
#	src/main/java/poomasi/domain/category/service/CategoryService.java
#	src/main/java/poomasi/domain/farm/dto/FarmResponse.java
#	src/main/java/poomasi/domain/farm/entity/Farm.java
#	src/main/java/poomasi/domain/farm/service/FarmFarmerService.java
#	src/main/java/poomasi/domain/farm/service/FarmService.java
#	src/main/java/poomasi/domain/product/_category/controller/CategoryAdminController.java
#	src/main/java/poomasi/domain/product/_category/controller/CategoryController.java
#	src/main/java/poomasi/domain/product/_category/dto/ProductListInCategoryResponse.java
#	src/main/java/poomasi/domain/product/_category/entity/Category.java
#	src/main/java/poomasi/domain/product/controller/ProductAdminController.java
#	src/main/java/poomasi/domain/product/controller/ProductController.java
#	src/main/java/poomasi/domain/product/controller/ProductFarmerController.java
#	src/main/java/poomasi/domain/product/dto/ProductRegisterRequest.java
#	src/main/java/poomasi/domain/product/dto/ProductResponse.java
#	src/main/java/poomasi/domain/product/entity/Product.java
#	src/main/java/poomasi/domain/product/service/ProductFarmerService.java
#	src/main/java/poomasi/domain/product/service/ProductService.java

stopmin committed:
feat: 충돌 해결

stopmin committed:
feat: 수정사항 삭제

Commit:
fix : bug 수정

amm0124 committed:
Merge pull request #48 from kakao-tech-campus-2nd-step3/feature/issue-47

fix : bug 수정

stopmin committed:
feat: S3PresignedUrl
- Presgiend Url 발급 메소드

stopmin committed:
refactor: AwsProperties 네이밍 수정

stopmin committed:
refactor: 패키지 분리
- aws/s3 분리
#32

jjt4515 committed:
S3 presignedUrl 테스트 코드 작성 #32

stopmin committed:
S3 설정 및 Presigned URL 관련 메소드 정의 (#51)

* feat: S3 Properties 설정
- S3Config 정의
- s3Presigner Bean으로 관리
#32

* feat: S3PresignedUrl
- Presgiend Url 발급 메소드
#32

* refactor: AwsProperties 네이밍 수정
#32

* refactor: 패키지 분리
- aws/s3 분리
#32

* refactor: EncryptionUtil 분리
#32

* S3 presignedUrl 테스트 코드 작성 #32

* feat: 테스트 API
#32

---------

Co-authored-by: 정진택 <87135698+jjt4515@users.noreply.github.com>

stopmin committed:
Merge pull request #52 from kakao-tech-campus-2nd-step3/week5

[Develop] 5주차 최종코드

amm0124 committed:
Merge pull request #53 from kakao-tech-campus-2nd-step3/develop

Develop

stopmin committed:
Merge branch 'week6' into feature/ISSUE-29

canyos committed:
feat: review 생성

canyos committed:
B
style: 아 맞다 정렬

canyos committed:
style: 아 맞다 정렬

canyos committed:
Merge remote-tracking branch 'origin/feature/ISSUE-29' into feature/ISSUE-29

canyos committed:
Merge pull request #43 from kakao-tech-campus-2nd-step3/feature/ISSUE-29

Review 만들기

canyos committed:
feat: review 생성

canyos committed:
Review 생성

Feature/issue 29

stopmin committed:
Merge branch 'week5' into week6

amm0124 committed:
Feature/issue 49 - 인증 및 인가 (#56)

* refactor: LoginRequest -> SignUpRequest 네임 변경 #39

* refactor: 회원가입 토큰생성 로직 삭제 #39

* refactor: 로그아웃 로직 삭제 #39

* chore: JwtUtil auth 폴더로 이동 #39

* JwtUtil 리팩토링 및 claims에 id, Type 추가 #39

* refactor: JwtUtil 리팩토링 #39

* feat : OAuth2 response interface 구현

* refactoring : member entity 변수 명 변경 및 builder 추가

* refactoring : member profile 기본 생성자 추가

* refactoring : member repository 메서드

* refactoring : jwt 검증 필터

* refactoring : 패키지 구조 변경을 위한 삭제

* refactoring : 로그아웃 필터 주석처리

* refactoring : User Detail 구현체 Oauth2User도 구현 받도록 변경

* feat : 회원가입 패키지 추가

* refactoring : username password 필터 리팩토링
- 메서드 변경

* refactoring : security 설정 변경

* feat : refresh token 관련 (reissue, refresh) 패키지 추가

* refactoring : 안 쓰는 request 삭제

* feat : blacklist 패키지 구조

* feat : 소셜 로그인 user detail service 구현 및 success handler(토큰 발급)

* feat : 회원가입 response dto 구현

* refactoring : jwt util 객체 변경

* delete : 안 쓰는 response import 삭제

* delete : 안 쓰는 response import 삭제

* refactoring : test를 위한 경로 open

* refactor: Member엔티티 리팩토링 #49

* refactor: MemberRepository 안쓰는 메소드 삭제 #49

* refactor: 토큰 reissue 리팩토링 #49

* refactor: 리프레시 토큰 저장 인터페이스 구현하여 분리  #49

* refactor: JwtUtil 수정 #49

* refactor: 블랙리스트 인터페이스 구현하여 분리 #49

* feat: jpa 토큰 만료 로직 추가 #49

* refactor: 레디스 에러 핸들링 공통화 #49

* feat: jpa 토큰 저장 스케쥴러 구현 #49

* feat: blacklist 및 refreshtoken config 설정 #49

* refactor: JwtUtil 수정 #49

* refactor: 빈 이름 중복 해결 #49

* Delete src/test/java/poomasi/global/config/s3 directory

* docs: build.gradle 중복 제거 #49

* refactor: 리프레시 토큰 key 변수 네임 변경 #49

* refactor:레디스, jpa 전환  프로파일 -> application 속성 이용으로 변경 #49

* fix: blacklist key->tokenKey로 변경 #49

* feat : username password authentication filter 완료

* feat : jwt filter 완성

* feat : jwt token을 위한 utility 수정

* feat : oauth2 응답을 위한 response dto 및 인터페이스 생성

* refactoring : member profile 변경

* refactoring : 회원가입 서비스 변경

* feat : 인가 확인 테스트 컨트롤러 작성

* feat : 로그아웃 성공 핸들러

* feat : oauth2 성공 핸들러

* refactoring : bean generator 수정 및 config 수정

* feat : logout filter 작성

* refactoring : username password 필터 수정
- username, password를 받아올 때, object mapper 사용

* refactoring : jwt util 수정

* refactoring : OAuth2 user detail 및 response 구현

* feat : 로그아웃 필터 작성

* refactoring : user detail get Member 추가

* refactoring : 회원가입 코드 변경

* refactoring : security bean, config 수정
- jwt util 의존성 추가

* feat : 로그아웃 handler 구현

* refactoring : jwt 필터 수정

* feat : test controller 구현

* conflict : conflict 해결 및 test controller 구현

* merge : master merge를 위한 commit. conflict 해결

* refactoring : redis global 이동

* refactoring : member profile column 추가
- 상세 주소, x좌표, y좌표

* refactoring : sout -> log 변경

* refactor: JwtUtil 약간 수정#49

* feat : 모든 경로에 대해서 security 적용 해제(임시)

---------

Co-authored-by: 정진택 <87135698+jjt4515@users.noreply.github.com>
Co-authored-by: amm0124 <amm012450@gmail.com>

stopmin committed:
예약 도메인을 생성한다  (#54)

* feat: Reservation 생성
- Farm
- Member
- FarmSchedule
#32

* refactor: Farm 도메인 일부 분리
#32

* feat: Reservation Service 구현
- get...메소드들
- 예약 생성 인터페이스
#32

* feat: 농장 예약 기능 구현
- 농장 예약
- 열린 농장인가?
- 유효한 농장인가?
- 유효한 스케쥴인가?
#31

* feat: 농장 조회
- 관리자나 자신의 농장만 조회 가능하도록 수정
#31

* feat: 농장 조회
- 관리자나 자신의 농장만 조회 가능하도록 수정
#31

* feat: 예약 취소
- 방문 3일 전까지 예약 취소 가능
#31

* fix: ADMIN은 가능하도록 수정
#31

* feature: 예약 조회
#31

stopmin committed:
Merge remote-tracking branch 'origin/week6' into week6

stopmin committed:
위시리스트를 생성한다 (#58)

* feat: product review 생성 조회

* feat: category - product 연관관계 추가

* feat: 카테고리 내 상품 조회

* fix: 빠진 파일 추가

* style: 아 맞다 정렬

* feat: 상품리뷰 - 상품리뷰사진 연관 관계 생성

* refactor: 평균 평점을 리뷰 추가할 때 계산하도록 변경

* feat: 충돌 해결

* feat: 수정사항 삭제

* feat: review 생성

* style: 아 맞다 정렬

* feature: 위시리스트 등록
- 재고와 무관하게 경우 위시리스트 가능하게
#57

* feature: 위시리스트 조회
#57

---------

Co-authored-by: canyos <4581974@naver.com>
Co-authored-by: canyos <31244128+canyos@users.noreply.github.com>

stopmin committed:
위시리스트를 생성한다 (#58)

* feat: product review 생성 조회

* feat: category - product 연관관계 추가

* feat: 카테고리 내 상품 조회

* fix: 빠진 파일 추가

* style: 아 맞다 정렬

* feat: 상품리뷰 - 상품리뷰사진 연관 관계 생성

* refactor: 평균 평점을 리뷰 추가할 때 계산하도록 변경

* feat: 충돌 해결

* feat: 수정사항 삭제

* feat: review 생성

* style: 아 맞다 정렬

* feature: 위시리스트 등록
- 재고와 무관하게 경우 위시리스트 가능하게

* feature: 위시리스트 조회

---------

Co-authored-by: canyos <4581974@naver.com>
Co-authored-by: canyos <31244128+canyos@users.noreply.github.com>

fix: JoinColumn 수정

stopmin committed:
Merge branch 'week6' of https://github.com/kakao-tech-campus-2nd-step3/Team10_BE into week6

amm0124 committed:
Merge pull request #60 from kakao-tech-campus-2nd-step3/week6

[Develop] 6주차 최종 코드 PR

amm0124 committed:
Merge pull request #61 from kakao-tech-campus-2nd-step3/develop

6주차 최종 코드

jjt4515 committed:
fix: aws 자격 증명 명시 #33

jjt4515 committed:
refactor: PresignedUrlPut 파라미터 수정 #33

jjt4515 committed:
feat: 이미지 정보 저장 로직 구현 #33

jjt4515 committed:
refactor: EncryptionUtil 인코딩 변수 추가 #33

jjt4515 committed:
refactor: presigned-url-put을 PostMapping으로 변경 #33

jjt4515 committed:
feat: 이미지 정보 관리 CRUD 작성 #33

jjt4515 committed:
refactor: 이미지 key 형식 수정 #33

jjt4515 committed:
fix: S3PresignedUrlController 오류 해결 #33

Commit:
fix : jwt filter 버그 수정
- 멤버 객체에서 id가 조회되지 않는 버그 수정

amm0124 committed:
Merge pull request #68 from kakao-tech-campus-2nd-step3/feature/ISSUE-65

jwt filter 버그 수정

canyos committed:
feat: category admin 권한 필요한거 확인

canyos committed:
feat: product farmer 권한 있는지 체크
product와 요청한 사람 id가 같은지도 체크

canyos committed:
feat: review 관련 member 권한 체크

canyos committed:
style: 아 맞다 정렬

canyos committed:
Merge remote-tracking branch 'origin/feature/ISSUE-69' into feature/ISSUE-69

stopmin committed:
feature: 팀 최대 인원/동일시간대 최대 예약 가능 팀 수 설정
#62

stopmin committed:
feature: 팀 최대 인원/동일시간대 최대 예약 가능 팀 수 설정
#62

stopmin committed:
feature: 팀 최대 인원/동일시간대 최대 예약 가능 팀 수 설정
- 생성자 수정
#62

stopmin committed:
feat: 시작시간/종료시간 추가
- entity 수정
#62

stopmin committed:
Merge remote-tracking branch 'origin/feature/ISSUE-62' into feature/ISSUE-62

stopmin committed:
feat: 농장 예약/예약취소 로직 수정
- 최대 수용가능 인원/동시간대 최대 수용 팀 수정
#62

jjt4515 committed:
feat: 이미지 삭제 url 수정

canyos committed:
refactor: admin권한 필요한거 몇개 추가

stopmin committed:
feat: 예약 가능 날짜 추가 API 수정
#62

stopmin committed:
chore: 에러문구 수정
#62

jjt4515 committed:
feat: 이미지 delete 문 오타 수정 #33

jjt4515 committed:
fix: 이미지 소프트 삭제 반영 #33

jjt4515 committed:
refactor: 이미지 서비스 리팩토링 #33

jjt4515 committed:
refactor: 소프트 삭제된 이미지 저장 시 복구시키는 로직 추가 #33

jjt4515 committed:
feat: 이미지 복구 로직 추가 #33

jjt4515 committed:
refactor: 이미지 복구 로직 수정 #33

jjt4515 committed:
Merge branch 'week8' of https://github.com/kakao-tech-campus-2nd-step3/Team10_BE into week8

jjt4515 committed:
feat: 멤버 조회 추가 #76

jjt4515 committed:
feat: 자격 증명 Dto 추가 #76

jjt4515 committed:
feat: 고객으로 전환 추가 #76

canyos committed:
feat : 장바구니 crud, test완료 (#66)

* feat : 장바구니 crud, test는 안해봄

* feat : order만들 때 사용할 쿼리 하나 추가

* refactor: test, dto만 분리함

jjt4515 committed:
Merge branch 'week8' into feature/ISSUE-76

Commit:
Merge remote-tracking branch 'refs/remotes/origin/feature/ISSUE-69' into feature/ISSUE-28

Commit:
build : 포트원(아임포트) 의존성 추가

Commit:
feat : ORDER, PAYMENT exception

Commit:
feat : security config - 허용 경로 변경

Commit:
feat : order entity 작성
- 나중에 농장 결제랑 연동 바로 할 수 있도록 공통 field는 추상 클래스로 작성하였습니다

Commit:
feat : 포트원 config 작성

Commit:
feat : 주문에 대한 상세 품목 entity

Commit:
feat : 기존 entity 연관관계 추가

Commit:
feat : order controller, service, repository 작성

Commit:
feat : dto 작성

Commit:
feat : 주문한 상품 repository 작성

Commit:
feat : 결제 entity , dto 작성
- PaymentMethod는 나중에 바뀔 수 예정

Commit:
feat : payment controller, service, repository 작성

Commit:
test : 포트원(아임포트) 테스트 작성

Commit:
feat : 환불 entity(미완)

amm0124 committed:
Merge pull request #83 from kakao-tech-campus-2nd-step3/feature/ISSUE-28

Feature/issue 28

amm0124 committed:
Merge pull request #84 from kakao-tech-campus-2nd-step3/week9

Week9 -> master

Commit:
Merge remote-tracking branch 'refs/remotes/origin/Review' into Master

# Conflicts:
#	build.gradle
#	src/main/java/poomasi/domain/auth/config/SecurityConfig.java
#	src/main/java/poomasi/domain/farm/entity/Farm.java
#	src/main/java/poomasi/domain/member/entity/Member.java
#	src/main/java/poomasi/domain/product/_category/controller/CategoryAdminController.java
#	src/main/java/poomasi/domain/product/_category/controller/CategoryController.java
#	src/main/java/poomasi/domain/product/controller/ProductFarmerController.java
#	src/main/java/poomasi/domain/product/dto/ProductRegisterRequest.java
#	src/main/java/poomasi/domain/product/entity/Product.java
#	src/main/java/poomasi/domain/product/service/ProductFarmerService.java
#	src/main/java/poomasi/domain/product/service/ProductService.java
#	src/main/java/poomasi/domain/review/controller/ReviewController.java
#	src/main/java/poomasi/domain/review/controller/farm/FarmReviewController.java
#	src/main/java/poomasi/domain/review/controller/product/ProductReviewController.java
#	src/main/java/poomasi/domain/review/dto/ReviewRequest.java
#	src/main/java/poomasi/domain/review/dto/ReviewResponse.java
#	src/main/java/poomasi/domain/review/entity/Review.java
#	src/main/java/poomasi/domain/review/service/ReviewService.java
#	src/main/java/poomasi/domain/review/service/farm/FarmReviewService.java
#	src/main/java/poomasi/domain/review/service/product/ProductReviewService.java
#	src/main/java/poomasi/global/error/BusinessError.java

jadie0319 commented on src/main/java/poomasi/domain/order/dto/response/OrderProductDetailsResponse.java:
```diff
@@ -0,0 +1,29 @@
+        ) {
+    public static OrderProductDetailsResponse fromEntity(OrderProductDetails orderProductDetails) {
+        return new OrderProductDetailsResponse(
+                orderProductDetails.getOrder().getId(),
```
order 를 조회하고 id 를 조회는 것보단 orderProductDetails.getOrderId() 같은 형태로 만드는건 어떻게 생각하시나요?

jadie0319 commented on src/main/java/poomasi/domain/order/_payment/service/PaymentService.java:
```diff
@@ -0,0 +1,149 @@
+    private final CartService cartService;
+
+    @Description("포트원 api 호출을 위한 accessToken 발급 메서드")
+    private String getPortOneAccessToken() throws IOException, IamportResponseException {
+        IamportResponse<AccessToken> authResponse = iamportClient.getAuth();
+        String accessToken = authResponse.getResponse().getToken();
+        return accessToken;
+    }
```
클래스 내부에서만 사용하는 private 메서드 보다는 외부에 공개되는 public 메서드가 더 중요한 것 같아요. public 메서드를 먼저 파악할 수 있게 위에 위치시키는건 어떨까요?

jadie0319 commented on src/main/java/poomasi/domain/product/_cart/repository/CartRepository.java:
```diff
@@ -0,0 +1,42 @@
+    void deleteAllByMemberId(Long memberId);
+
+    @Modifying
+    @Transactional
```
repository 메서드에 `@Transactional` 어노테이션은 없어도 될 것 같아요.

jadie0319 commented on src/main/java/poomasi/domain/product/_cart/dto/CartRegisterRequest.java:
```diff
@@ -0,0 +1,21 @@
+        Integer count
+) {
+
+    public Cart toEntity(Member member) {
+        return Cart.builder()
+                .memberId(member.getId())
+                .productId(productId)
+                .selected(Boolean.TRUE)
+                .count(count != null ? count : 1)
+                .build();
+    }
```
Cart 생성 로직이 DTO 에 있는 것은 좀 어색한것 같아요. Service 에서 카트를 만들어보면 어떨까요?
DTO 가 엔티티 생성 책임을 가져야 하는지도 팀원과 논의 해보세요 :)

jadie0319 commented on src/main/java/poomasi/domain/product/_cart/service/CartService.java:
```diff
@@ -0,0 +1,131 @@
+        }
+
+        Cart cart = cartRequest.toEntity(member);
+        if (product.getStock() < cart.getCount()) {
+            throw new BusinessException(BusinessError.PRODUCT_STOCK_ZERO);
+        }
```
카트를 생성하기 위해선 이 조건이 통과해야 생성이 가능하네요.
이 조건은 중요한 도메인 규칙 같아요. 도메인 객체에서 해당 조건을 검증하게 바꿔보면 어떨까요?

jadie0319 commented on src/main/java/poomasi/domain/product/_cart/service/CartService.java:
```diff
@@ -0,0 +1,131 @@
+    private final ProductRepository productRepository;
+
+    @Transactional
+    public Long addCart(CartRegisterRequest cartRequest) {
```
addCart 라는 메서드명을 보고 카트에 물건을 넣는 로직을 상상했는데, 내용은 카트를 등록?생성하는 기능인 것 같아요. 카트를 생성하는 의미를 가진 메서드명으로 바꿔보면 어떨까요...?

jadie0319 commented on src/main/java/poomasi/domain/review/entity/Review.java:
```diff
@@ -42,22 +46,26 @@ public class Review {
     @Enumerated(EnumType.STRING)
     private EntityType entityType;
```
EntityType 의 Entity 가 자바 엔티티 객체를 뜻하는 그 엔티티 인가요?
작성된 리뷰가 어디에 작성된지를 구분하기 위한 값 같은데,
Enity 보다는 비즈니스를 표현할 수 있는 다른 단어를 고민해보시면 좋을 것 같아요.

jadie0319 commented on src/main/java/poomasi/domain/order/_payment/service/PaymentService.java:
```diff
@@ -0,0 +1,149 @@
+        Order order = orderRepository.findByMerchantUid(merchantUid)
+                .orElseThrow(() -> new BusinessException(ORDER_NOT_FOUND));
+        
+        if(order.getOrderStatus()!=PENDING){ //이미 처리한 주문이라면
```
order 의 상태를 조회해서 PENDING 인지 아닌지 묻는 것 보다는 order 에게 시켜보면 어떨까요?
예를 들면, order.isPending() 과 같이 표현할 수 있을 것 같아요. 
객체로부터 데이터를 조회해서 로직을 서비스단에 구현하는 것 보다 객체에 로직을 위치 시키면 도메인 객체가 하는 일을 파악하기 쉽고 적절한 단어로 표현할 수 있는 장점이 있어요.
`묻지말고 시켜라. 라는 객체지향 원칙에 대해서도 검색해보시면 도움이 될 것 같아요.

amm0124 replied:
넵!! 알겠습니다!!

amm0124 replied:
order와 payment의 도메인을 어떻게 나눠야 할 지 고민이 들었습니다 ㅠ.ㅠ

jadie0319 commented on src/main/java/poomasi/domain/product/_cart/dto/CartRegisterRequest.java:
```diff
@@ -0,0 +1,21 @@
+import poomasi.domain.member.entity.Member;
+import poomasi.domain.product._cart.entity.Cart;
+
+@Slf4j
```
해당 어노테이션이 필요할까요??

jadie0319 commented on src/main/java/poomasi/domain/review/dto/ReviewResponse.java:
```diff
@@ -15,7 +15,7 @@ public static ReviewResponse fromEntity(Review review) {
                 review.getId(),
                 review.getEntityId(),
-                //productReview.getReviewer().getId(),
+                review.getReviewer().getMemberProfile().getName(),
```
review.getReviewerName() 과 같은 메서드를 만들어 볼수 있을것 같아요 :)

jadie0319 commented on src/main/java/poomasi/domain/auth/config/SecurityConfig.java:
```diff
@@ -114,14 +117,14 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
+
         http.oauth2Login(AbstractHttpConfigurer::disable);
 
         CustomUsernamePasswordAuthenticationFilter customUsernameFilter =
```
134,135 Line 에서 UsernamePasswordAuthenticationFilter 와 같은 우선순위, 또는 이전에 실행되는 필터를 addFilterAt 과 addFilterBefore 메서드를 통해 지정해주고 있어요.

그런데 CustomUsernamePasswordAuthenticationFilter 는 UsernamePasswordAuthenticationFilter 를 상속받았어요.
UsernamePasswordAuthenticationFilter 와 CustomUsernamePasswordAuthenticationFilter 를 같은 우선순위로 실행( addFilterAt 메서드 사용 )하게 했는데 동일한 타입(상속받았으니) 의 필터를 동일한 우선순위로 사용하게 되어 두번 호출된것처럼 된게 아닐까 하는 의심이 듭니다.

보다 정확한 원인 파악은 디버그모드로 실행해 로그를 확인해봐야 파악할 수 있을것 같습니다.

아니면 필터를 하나만 적용해보면서 테스트해보면 어떨까요??
```java
//        CustomUsernamePasswordAuthenticationFilter customUsernameFilter =
//                new CustomUsernamePasswordAuthenticationFilter(authenticationManager(authenticationConfiguration), jwtUtil);
//        customUsernameFilter.setFilterProcessesUrl("/api/login");
//
//        http.addFilterAt(customUsernameFilter, UsernamePasswordAuthenticationFilter.class);
        http.addFilterBefore(new JwtAuthenticationFilter(jwtUtil, userDetailsService), UsernamePasswordAuthenticationFilter.class);
```
이렇게 주석처리하고 실행했을땐 두번 호출이 될까요?

amm0124 replied:
아!! 이해했습니다!! 해보겠습니다

jadie0319 reviewed (approved):
안녕하세요!
중간고사와 병행하시면서 9주차 코드 PR 올리시느라 고생하셨습니다 👍
다음주부터 날씨가 추워진다는데 건강관리 잘 하면서 프로젝트 진행하시기 바랍니다~
질문이나 궁금하신점 있으시면 DM 이나 멘토링 신청해주세요
화이팅 하세요!

질문해주신것 답변 드릴게요!

### 요청을 하면 응답이 2번 온다는 경우가 존재한다고 하는데, security 부근에서 일어나는 문제 같다고 하는데 왜 그런지 질문 드립니다!

현재 프로젝트를 실행할 수 없는데, 정확한 원인을 찾으려면 프로젝트를 디버그 모드로 실행하고 breakpoint 를 찍으면서 추적해봐야 알 것 같습니다. 두번의 요청이 모두 동일하게 같은 로직을 타는지도 확인을 해봐야 할 것 같네요. 예상으로는 말씀하신것과 같이 스프링 시큐리티 부분의 설정때문에 발생하는 이슈인 것 같아요. 관련해서 SecurityConfig 에 있는 코드리뷰 확인 부탁드려요.

### test code 작성 시 팁?!을 듣고 싶습니다. 커버리지 툴을 도입하자는 의견이 나왔는데 이 부분에 대해서도 어떤지 궁금합니다!

커버리지 툴을 도입하면 어떤 로직을 테스트 했는지 또는 테스트에서 실행하지 않은 로직을 파악하는데 도움이 될거에요. 하지만 **더 중요한 것은 테스트 코드를 작성**하는 것이에요. 테스트 코드를 잘 작성하고 관리하는 방법을 공부하고, 실제로 테스트코드를 만들어야 합니다. 그 이후에  커버리지 툴 도입을 고민해도 된다고 생각합니다!

**도메인 객체**를 테스트 하려면 일단 비즈니스 로직이 도메인 객체에 있어야 해요. 서비스단에 있는 비즈니스 로직을 도메인 객체로 옮기고 테스트 코드를 작성하는 연습을 해보시면 될 것 같아요. 

**서비스**를 테스트 하려면 **테스트 환경을 격리**시켜야 합니다. 예를 들어 결제 API 호출과 같은 로직을 테스트 코드에서 실제로 호출한다고 생각해볼까요? 테스트 코드를 실행할 때마다 결제 api 를 호출해야 하니 네트워크 환경에 따라 테스트 속도가 느려질 것 같아요. 그리고 만약 결제 서비스가 제대로 동작하지 않으면 서비스 코드는 반드시 실패하게 되겠죠. 

이런 외부 서비스의 영향을 받지 않게 하기 위하여 **테스트 코드를 격리** 시켜야 합니다. Mock 객체에 대해서 검색해보시면 도움이 될 것 같습니다.

### 품목을 구매할 때 (사전주문->외부 api 결제 -> 주문 완료) 형태로 구현하려고 하는데, 주문 완료 메서드를 transaction으로 처리하려고 했습니다. 이 때 동시성 처리는 어떻게 처리해야 하는지 질문을 드리고 싶습니다!

해당 로직에서는 고려할 필요가 있을까 생각이 듭니다. 한명의 사용자가 주문을 생성하고 하나의 주문이 완료되기 때문에 동시성 관련해서 문제가 발생할 것 같지 않아요. 예를 들면 특정 테이블에 있는 어떤 데이터를 **여러 사용자가 동시에 변경**하려고 할 때 동시성 이슈가 발생해요. 해당 로직에서는 그런 부분이 없는 것 같아요. 추가로 동시성 관련한 문제를 공부하시려면 낙관적락, 비관적락에 대해서 검색해 보시면 자료를 많이 얻으실 수 있어요. 🙂

stopmin committed:
feat: 농장 체험 가능 스케쥴 추가 API
#62

stopmin committed:
Merge branch 'Master' into feature/ISSUE-62

stopmin committed:
feat: 농장 스케쥴 추가 검증 수정
#62

stopmin committed:
feat: 농장 최대 인원 & 예상 금액
#62

stopmin committed:
feat: 요청시 일단 수락하도록 설정
#62

stopmin committed:
Farm Schedule 관리 방식 및 금액제도 수정 (#73)

* feature: 팀 최대 인원/동일시간대 최대 예약 가능 팀 수 설정
#62

* feature: 팀 최대 인원/동일시간대 최대 예약 가능 팀 수 설정
#62

* feature: 팀 최대 인원/동일시간대 최대 예약 가능 팀 수 설정
- 생성자 수정
#62

* feat: 시작시간/종료시간 추가
- entity 수정
#62

* feat: 농장 예약/예약취소 로직 수정
- 최대 수용가능 인원/동시간대 최대 수용 팀 수정
#62

* feat: 예약 가능 날짜 추가 API 수정
#62

* chore: 에러문구 수정
#62

* feat: 농장 체험 가능 스케쥴 추가 API
#62

* feat: 농장 스케쥴 추가 검증 수정
#62

* feat: 농장 최대 인원 & 예상 금액
#62

* feat: 요청시 일단 수락하도록 설정
#62

canyos committed:
기존 category, product, review 코드 security 적용하기

canyos committed:
feat: product tag 생성 (#72)

* feat: product tag 생성

* refactor: entity빼고 enum으로 관리하게 변경

* style: 정렬

* fix: fetchtype변경, 변수명 변경, 공통로직 추출

canyos committed:
feat: 인증인가 추가 (#74)

stopmin committed:
Merge branch 'feature/ISSUE-62' into week10

# Conflicts:
#	src/main/java/poomasi/domain/farm/_schedule/dto/FarmScheduleUpdateRequest.java
#	src/main/java/poomasi/domain/farm/_schedule/entity/FarmSchedule.java
#	src/main/java/poomasi/domain/farm/_schedule/repository/FarmScheduleRepository.java
#	src/main/java/poomasi/domain/farm/_schedule/service/FarmScheduleService.java

stopmin committed:
bug: Merge 잘못된 부분 정정

stopmin committed:
bug: Merge 잘못된 부분 정정

jjt4515 committed:
Merge branch 'week10' into feature/ISSUE-33

stopmin committed:
FarmSchedule 누락 사항 (#95)

* feat: 스케쥴 존재하는지 확인하도록 수정
#92

* chore: 필요없는 에러 제거
#92

* feat: 스케쥴 ID도 전달하도록 수정
#92

* chore: 시큐리티 수정
#92

jjt4515 committed:
Merge pull request #67 from kakao-tech-campus-2nd-step3/feature/ISSUE-33

이미지 관리 테이블 작성한다

jjt4515 committed:
Merge branch 'week10' into feature/ISSUE-76

jjt4515 committed:
Merge pull request #79 from kakao-tech-campus-2nd-step3/feature/ISSUE-76

회원 조회

stopmin committed:
예약 & 농장 조회 Security 적용 (#98)

stopmin committed:
[TEST] Mockito 설정 및 기본 테스트 코드 추가 (#102)

* build: Mockito 의존성 추가
- 5.12.0 Version
#101

* test: 농장 조회 테스트 추가
- 테스트 용입니당..!
#101

* chore: 의존성 정리
#101

stopmin committed:
[CICD] ECS CICD 파이프라인 구축 (#105)

* feat: Dockerfile정의
#103

* feat: HealthcheckController 정의
- 200 ok
#103

* build: jar plain 생성 x
#103

* feat: prod 환경 추가
- application-prod.yml 추가
#103

* feat: deploy-ecs.yml 정의
#103

* feat: deploy-ecs.yml 수정
- build-arg 수정
#103

stopmin committed:
Merge pull request #106 from kakao-tech-campus-2nd-step3/week10

10주차 코드 중간 PR - deploy-ecs.yml test
